_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,281 @@
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+    // deploys KafkaBridge consumer client
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // create instance of Kafka client with specific configuration
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    // send messages to Kafka + verification
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    // waiting until KafkaBridge consumer client successfully end
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    
+    // deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
+}
+```
+
+As you can see, this test case ensures that we are able to receive messages from the `KafkaBridge` component.
+In the setup phase we deploy `cluster-operator`, a single one node Kafka cluster and also the `KafkaBridge` itself.
+Everything is deployed into a real Kubernetes cluster.
+All test cases with related test suites can be found in the [systemtest](https://github.com/strimzi/strimzi-kafka-operator/tree/master/systemtest) 
+module.
+Also worth mentioning is that our system tests currently take around 30 hours. 
+We have 3 separate sub-sets of tests, which run in parallel so it is "only" approximately 10 hours per each sub-set.
+Additionally, there is a mvn profile for the main groups - `acceptance`, `regression`, `smoke`, `bridge` and `all`,
+but we suggest to use a profile with id `all` (default) and then include or exclude specific groups.
+If you want specify the profile, Strimzi uses maven in its build and you can run particular test groups using a maven 
+profile. 
+Here is an example of how we trigger some profiles:
+
+```
+Generic definition
+
+mvn -f ${workspace}/systemtest/pom.xml -P all verify -Dgroups=${testProfile} -DexcludedGroups=${excludeGroups}
+
+Acceptance profile
+
+mvn -f ${workspace}/systemtest/pom.xml -P all verify -Dgroups=acceptance -DexcludedGroups=""
+```
```

Comment:
This looks weird on the website with the text inside the code. I think ideally you should take it out and have the code formatting wrap only the `mvn` command. Or at least remove the empty lines and prefix the text with `# ` as comment.